server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -716,13 +722,21 @@ private void validateIndexTemplateV2(String name, ComposableIndexTemplate indexT
             indexTemplate.metadata(),
             indexTemplate.getDataStreamTemplate(),
             indexTemplate.getAllowAutoCreate(),
-            indexTemplate.getIgnoreMissingComponentTemplates()
+            indexTemplate.getIgnoreMissingComponentTemplates(),
+            indexTemplate.deprecated()
         );
 
         validate(name, templateToValidate);
         validateDataStreamsStillReferenced(currentState, name, templateToValidate);
         validateLifecycleIsOnlyAppliedOnDataStreams(currentState.metadata(), name, templateToValidate);
 
+        if (templateToValidate.deprecated() == false) {
+            validateUseOfDeprecatedComponentTemplates(name, templateToValidate, currentState.metadata().componentTemplates());
+            validateUseOfDeprecatedIngestPipelines(name, currentState.metadata().custom(IngestMetadata.TYPE), combinedSettings);
+            // we don't have access to the core/main plugin here so we can't use the IndexLifecycleMetadata type
+            // validateUseOfDeprecatedIlmPolicies(name, currentState.metadata().custom(IndexLifecycleMetadata.TYPE), combinedSettings);
```

Comment:
@dakrone any idea how we should validate ILM policies? Should we just leave it out but still validate component templates and ingest pipelines?

Reply:
I don't think we'll be able to easily do that validation here. I think we should come up with a separate plan for the best option for deprecation logging of deprecated ILM policies. I wouldn't block this PR on it.

Reply:
I think we can remove this commented out call and have it be a regular TODO in the code, since we won't be able to validate that right here for now anyway.